### DISCUSSION TITLE
Can't do parallel queries

### DISCUSSION THREAD

**Discussion by JuanM04** [prisma/prisma#3319] [Q&A] [answered]
I have a website. This website makes two queries in parallel to the API, and this one gets data from the DB using Prisma.

Always, a 100% of the time, one of the queries fails, throwing `Cannot read property 'signalCode' of undefined`. If I remove one of the queries, no matter which, all works, but I can't make both work at the same time.

Anyone knows why this happens? or how to solve it?

**Comment by janpio** [comment, marked as answer]
How are you calling Prisma Client in your code?

**Reply by JuanM04** [reply]
```js
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

export default async (req, res) => {
  const { table } = req.body;

  try {
    if (table === "clients") {
      res.send(
        await prisma.client.findMany({
          take: 7,
          orderBy: {
            id: "desc",
          },
          include: {
            carrier: true,
          },
        })
      );
    } else {
      res.send(
        await prisma.carrier.findMany({
          take: 7,
          orderBy: {
            id: "desc",
          },
        })
      );
    }
  } catch (error) {
    console.error(error);
    res.status(500).send("Unknown error");
  } finally {
    await prisma.$disconnect();
  }
};
```

**Reply by ryands17** [reply]
Hey @JuanM04 👋 
This looks like you're executing in a serverless environment. I would suggest removing `prisma.$disconnect` and trying again, as for subsequent queries, the already open connections can be used. If this still produces an issue, open a bug report [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) mentioning the reproduction steps and the environment that you're deploying to.

**Reply by JuanM04** [reply]
It worked, you are a genius!
By the way, I'm not running it serverless, it's just Next.js running with `next start`. If I create a new `PrismaClient` in each API endpoint, does Prisma use the same connection, or I need to use the same `PrismaClient` for all my endpoints?

**Reply by janpio** [reply]
If  this is a reproducible problem, we should have a bug issue about this. You should not be able to so easily shoot yourself in the foot. Can you maybe create a minimal reproduction in a repo for this @JuanM04?

**Reply by JuanM04** [reply]
I've been trying but I can't reproduce it. I don't know why.

**Reply by josgraha** [reply]
fwiw, i think next api calls are [serverless containers](https://nextjs.org/docs/api-reference/next.config.js/build-target#serverless-target) by default anyway

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
